build error by Result and Returning multiple values

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relevant modules using `v outdated` and `v install` -->

<!-- You can use `v doctor` to fill up the next fields -->
**V version: 0.3.0 584c035** 
**OS: ArchLinux** 

<!-- Please include the standalone minimal reproducible code.
If not, it may be closed by the administrators. -->
**What did you do?**
```v
struct Err{
	msg  string
	code int
}

fn (e Err) msg() string{
	return e.msg
}

fn (e Err) code() int{
	return e.code
}

fn foo() ?string{
	return 'foo'
}

fn bar() !(string, int){
	a := foo() or {
		return IError(Err{ msg: 'error test' })
	}
	return a, 1
}

fn main(){
	b, _ := bar() or { panic(err) }
}
```

**What did you expect to see?**
No Error.
 
**What did you see instead?**
```
==================
/usr/lib64/crt1.o: error: Invalid relocation entry [15] '.rela.debug_info' @ 000004f5
tcc: error: file 'crt1.o' not found
/tmp/v_1000/bug.6359441838897102903.tmp.c:12160: error: '{' expected (got ";")
...
==================
```

--- a/vlib/v/gen/c/assign.v
+++ b/vlib/v/gen/c/assign.v
@@ -508,9 +508,9 @@ fn (mut g Gen) assign_stmt(node_ ast.AssignStmt) {
 fn (mut g Gen) gen_multi_return_assign(node &ast.AssignStmt, return_type ast.Type) {
 	// multi return
 	// TODO Handle in if_expr
-	is_opt := return_type.has_flag(.optional)
+	is_opt := return_type.has_flag(.optional) || return_type.has_flag(.result)
 	mr_var_name := 'mr_$node.pos.pos'
-	mr_styp := g.typ(return_type.clear_flag(.optional))
+	mr_styp := g.typ(return_type.clear_flag(.optional).clear_flag(.result))
 	g.write('$mr_styp $mr_var_name = ')
 	g.expr(node.right[0])
 	g.writeln(';')

--- a/vlib/v/tests/results_multi_return_test.v
+++ b/vlib/v/tests/results_multi_return_test.v
@@ -1,0 +1,29 @@
+struct Err {
+	msg  string
+	code int
+}
+
+fn (e Err) msg() string {
+	return e.msg
+}
+
+fn (e Err) code() int {
+	return e.code
+}
+
+fn foo() ?string {
+	return 'foo'
+}
+
+fn bar() !(string, int) {
+	a := foo() or { return IError(Err{
+		msg: 'error test'
+	}) }
+	return a, 1
+}
+
+fn test_results_multi_return() {
+	b, _ := bar() or { panic(err) }
+	println(b)
+	assert b == 'foo'
+}

